=== modified file 'scripts/mysql_install_db.pl.in'
--- scripts/mysql_install_db.pl.in	2013-02-28 13:32:03 +0000
+++ scripts/mysql_install_db.pl.in	2013-07-02 09:46:23 +0000
@@ -428,35 +428,35 @@
 # This will be the default config file
 # ----------------------------------------------------------------------
 
-$config_file= "$basedir/my.cnf";
-
-my $cfg_template= find_in_basedir($opt,"file","my-default.cnf",
-				  "share","share/mysql","support-files");
--e $cfg_template or cannot_find_file("my-default.cnf");
-
-$copy_cfg_file= $config_file;
+#$config_file= "$basedir/my.cnf";
+
+#my $cfg_template= find_in_basedir($opt,"file","my-default.cnf",
+#				  "share","share/mysql","support-files");
+#-e $cfg_template or cannot_find_file("my-default.cnf");
+my  $cfg_template = "/usr/share/mysql/my-default.cnf";
+#$copy_cfg_file= $config_file;
 my $failed_write_cfg= 0;
-if (-e $copy_cfg_file)
-{
-  $copy_cfg_file =~ s/my.cnf/my-new.cnf/;
-  # Too early to print warning here, the user may not notice
-}
-open (TEMPL, $cfg_template) or error($opt, "Could not open config template $cfg_template");
-if (open (CFG, "> $copy_cfg_file"))
-{
-  while (<TEMPL>)
-  {
-    # Remove lines beginning with # *** which are template comments
-    print CFG $_ unless /^# \*\*\*/;
-  }
-  close CFG;
-}
-else
-{
-  warning($opt,"Could not write to config file $copy_cfg_file: $!");
-  $failed_write_cfg= 1;
-}
-close TEMPL;
+#if (-e $copy_cfg_file)
+#{
+#  $copy_cfg_file =~ s/my.cnf/my-new.cnf/;
+#  # Too early to print warning here, the user may not notice
+#}
+#open (TEMPL, $cfg_template) or error($opt, "Could not open config template $cfg_template");
+#if (open (CFG, "> $copy_cfg_file"))
+#{
+#  while (<TEMPL>)
+#  {
+#    # Remove lines beginning with # *** which are template comments
+#    print CFG $_ unless /^# \*\*\*/;
+#  }
+#  close CFG;
+#}
+#else
+#{
+#  warning($opt,"Could not write to config file $copy_cfg_file: $!");
+#  $failed_write_cfg= 1;
+#}
+#close TEMPL;
 
 # ----------------------------------------------------------------------
 # Now we can get arguments from the groups [mysqld] and [mysql_install_db]

