diff -Nur mysql-test/t/disabled.def.orig mysql-test/t/disabled.def
--- mysql-test/t/disabled.def.orig      2010-01-15 12:46:09.000000000 -0500
+++ mysql-test/t/disabled.def   2010-01-31 14:09:18.000000000 -0500
@@ -11,3 +11,8 @@
 ##############################################################################
 kill                     : Bug#37780 2008-12-03 HHunger need some changes to be robust enough for pushbuild.
 query_cache_28249        : Bug#43861 2009-03-25 main.query_cache_28249 fails sporadically
+read_many_rows_innodb    : Bug#433409 2009-09-20 the test fails on 5.1.37 https://bugs.launchpad.net/bugs/433409
+log_tables-big           : Bug#48646 2009-12-12 mysql-test main.log_tables-big fails on 5.1.40
+information_schema       : Conflict with 5.1-percona-patch
+information_schema_db    : Conflict with 5.1-percona-patch
+mysqlshow                : Conflict with 5.1-percona-patch
