--- mysql-test/t/disabled.def.orig      2009-12-16 13:21:00.000000000 -0500
+++ mysql-test/t/disabled.def   2010-01-03 11:58:54.000000000 -0500
@@ -13,3 +13,5 @@
 query_cache_28249        : Bug#43861 2009-03-25 main.query_cache_28249 fails sporadically
 rpl_killed_ddl           : Bug#45520: rpl_killed_ddl fails sporadically in pb2
 innodb-autoinc           : Bug#49267 2009-12-02 test fails on windows because of different case mode
+read_many_rows_innodb    : Bug#433409 2009-09-20 the test fails on 5.1.37 https://bugs.launchpad.net/bugs/433409
+log_tables-big           : Bug#48646 2009-12-12 mysql-test main.log_tables-big fails on 5.1.40
