--- mysql-test/t/events_stress.test.orig	2009-07-05 10:29:14.000000000 +0000
+++ mysql-test/t/events_stress.test	2009-07-05 10:30:49.000000000 +0000
@@ -61,6 +61,7 @@
 }
 --enable_query_log
 SELECT COUNT(*) FROM INFORMATION_SCHEMA.EVENTS WHERE EVENT_SCHEMA='events_conn1_test2';
+SET @old_event_scheduler=@@event_scheduler;
 SET GLOBAL event_scheduler=on;
 --sleep 2.5
 DROP DATABASE events_conn1_test2;
@@ -137,5 +138,5 @@
 DROP DATABASE events_test;
 
 # Cleanup
-SET GLOBAL event_scheduler=off;
+SET GLOBAL event_scheduler=@old_event_scheduler;
 --source include/check_events_off.inc
--- mysql-test/r/events_stress.result.orig	2009-07-05 10:54:30.000000000 +0000
+++ mysql-test/r/events_stress.result	2009-07-05 10:54:48.000000000 +0000
@@ -32,6 +32,7 @@
 SELECT COUNT(*) FROM INFORMATION_SCHEMA.EVENTS WHERE EVENT_SCHEMA='events_conn1_test2';
 COUNT(*)
 50
+SET @old_event_scheduler=@@event_scheduler;
 SET GLOBAL event_scheduler=on;
 DROP DATABASE events_conn1_test2;
 SET GLOBAL event_scheduler=off;
@@ -63,4 +64,4 @@
 DROP TABLE fill_it2;
 DROP TABLE fill_it3;
 DROP DATABASE events_test;
-SET GLOBAL event_scheduler=off;
+SET GLOBAL event_scheduler=@old_event_scheduler;
