--- mysql-test/r/mysqlshow.result.orig  2009-11-09 15:17:43.000000000 -0800
+++ mysql-test/r/mysqlshow.result       2009-11-09 15:17:53.000000000 -0800
@@ -107,6 +107,21 @@
 | TRIGGERS                              |
 | USER_PRIVILEGES                       |
 | VIEWS                                 |
+| INNODB_BUFFER_POOL_PAGES_INDEX        |
+| INNODB_RSEG                           |
+| INNODB_LOCKS                          |
+| INNODB_BUFFER_POOL_PAGES              |
+| XTRADB_ENHANCEMENTS                   |
+| INNODB_TRX                            |
+| XTRADB_ADMIN_COMMAND                  |
+| INNODB_LOCK_WAITS                     |
+| INNODB_CMP_RESET                      |
+| INNODB_CMP                            |
+| INNODB_CMPMEM_RESET                   |
+| INNODB_TABLE_STATS                    |
+| INNODB_CMPMEM                         |
+| INNODB_INDEX_STATS                    |
+| INNODB_BUFFER_POOL_PAGES_BLOB         |
 +---------------------------------------+
 Database: INFORMATION_SCHEMA
 +---------------------------------------+
@@ -140,6 +155,21 @@
 | TRIGGERS                              |
 | USER_PRIVILEGES                       |
 | VIEWS                                 |
+| INNODB_BUFFER_POOL_PAGES_INDEX        |
+| INNODB_RSEG                           |
+| INNODB_LOCKS                          |
+| INNODB_BUFFER_POOL_PAGES              |
+| XTRADB_ENHANCEMENTS                   |
+| INNODB_TRX                            |
+| XTRADB_ADMIN_COMMAND                  |
+| INNODB_LOCK_WAITS                     |
+| INNODB_CMP_RESET                      |
+| INNODB_CMP                            |
+| INNODB_CMPMEM_RESET                   |
+| INNODB_TABLE_STATS                    |
+| INNODB_CMPMEM                         |
+| INNODB_INDEX_STATS                    |
+| INNODB_BUFFER_POOL_PAGES_BLOB         |
 +---------------------------------------+
 Wildcard: inf_rmation_schema
 +--------------------+
