--- mysql-test/r/mysqlbinlog_row_big.result.orig	2009-04-17 18:58:20.000000000 +0900
+++ mysql-test/r/mysqlbinlog_row_big.result	2009-04-17 18:56:48.000000000 +0900
@@ -1,4 +1,9 @@
 #
+# We need big packets.
+#
+SET @old_global_max_allowed_packet=@@global.max_allowed_packet;
+SET @@global.max_allowed_packet= 1024*1024*1024;
+#
 # Preparatory cleanup.
 #
 DROP TABLE IF EXISTS t1;
@@ -7,10 +12,6 @@
 #
 SET timestamp=1000000000;
 #
-# We need big packets.
-#
-SET @@global.max_allowed_packet= 1024*1024*1024;
-#
 # Delete all existing binary logs.
 #
 RESET MASTER;
@@ -71,4 +72,5 @@
 # Cleanup.
 #
 DROP TABLE t1;
+SET @@global.max_allowed_packet=@old_global_max_allowed_packet;
 remove_file $MYSQLTEST_VARDIR/tmp/mysqlbinlog_big_1.out
--- mysql-test/t/mysqlbinlog_row_big.test.orig	2009-04-17 18:57:58.000000000 +0900
+++ mysql-test/t/mysqlbinlog_row_big.test	2009-04-17 18:56:32.000000000 +0900
@@ -24,6 +24,16 @@
 --source include/big_test.inc
 
 --echo #
+--echo # We need big packets.
+--echo #
+connect (con1, localhost, root,,);
+connection con1;
+SET @old_global_max_allowed_packet=@@global.max_allowed_packet;
+SET @@global.max_allowed_packet= 1024*1024*1024;
+connect (con2, localhost, root,,);
+connection con2;
+
+--echo #
 --echo # Preparatory cleanup.
 --echo #
 --disable_warnings
@@ -36,11 +46,6 @@
 SET timestamp=1000000000;
 
 --echo #
---echo # We need big packets.
---echo #
-SET @@global.max_allowed_packet= 1024*1024*1024;
-
---echo #
 --echo # Delete all existing binary logs.
 --echo #
 RESET MASTER;
@@ -122,6 +127,9 @@
 --echo #
 DROP TABLE t1;
 
+connection con1;
+SET @@global.max_allowed_packet=@old_global_max_allowed_packet;
+
 --echo remove_file \$MYSQLTEST_VARDIR/$mysqlbinlog_output
 #
 # NOTE: If you want to see the *huge* mysqlbinlog output, disable next line:
