--- mysql-test/r/variables-big.result.orig	2009-04-17 19:18:57.000000000 +0900
+++ mysql-test/r/variables-big.result	2009-04-17 19:17:10.000000000 +0900
@@ -1,20 +1,21 @@
+set @pid_temp = (select ID from information_schema.processlist);
 set session transaction_prealloc_size=1024*1024*1024*1;
-show processlist;
-Id	User	Host	db	Command	Time	State	Info
-1	root	localhost	test	Query	0	NULL	show processlist
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
+TRUE
+1
 set session transaction_prealloc_size=1024*1024*1024*2;
-show processlist;
-Id	User	Host	db	Command	Time	State	Info
-1	root	localhost	test	Query	0	NULL	show processlist
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
+TRUE
+1
 set session transaction_prealloc_size=1024*1024*1024*3;
-show processlist;
-Id	User	Host	db	Command	Time	State	Info
-1	root	localhost	test	Query	0	NULL	show processlist
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
+TRUE
+1
 set session transaction_prealloc_size=1024*1024*1024*4;
-show processlist;
-Id	User	Host	db	Command	Time	State	Info
-1	root	localhost	test	Query	0	NULL	show processlist
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
+TRUE
+1
 set session transaction_prealloc_size=1024*1024*1024*5;
-show processlist;
-Id	User	Host	db	Command	Time	State	Info
-1	root	localhost	test	Query	0	NULL	show processlist
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
+TRUE
+1
--- mysql-test/t/variables-big.test.orig	2009-04-17 19:18:12.000000000 +0900
+++ mysql-test/t/variables-big.test	2009-04-17 19:16:26.000000000 +0900
@@ -8,13 +8,14 @@
 # Bug #27322  	failure to allocate transaction_prealloc_size causes crash
 #
 
+set @pid_temp = (select ID from information_schema.processlist);
 set session transaction_prealloc_size=1024*1024*1024*1;
-show processlist;
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
 set session transaction_prealloc_size=1024*1024*1024*2;
-show processlist;
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
 set session transaction_prealloc_size=1024*1024*1024*3;
-show processlist;
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
 set session transaction_prealloc_size=1024*1024*1024*4;
-show processlist;
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
 set session transaction_prealloc_size=1024*1024*1024*5;
-show processlist;
+select @pid_temp = (select ID from information_schema.processlist) as 'TRUE';
