diff -r -c --exclude=*.info* glibc-2.2.4/linuxthreads/internals.h glibc-2.2.4-new/linuxthreads/internals.h
*** glibc-2.2.4/linuxthreads/internals.h	Mon Jul 23 20:54:13 2001
--- glibc-2.2.4-new/linuxthreads/internals.h	Tue Apr 16 15:08:03 2002
***************
*** 339,345 ****
     THREAD_SELF implementation is used, this must be a power of two and
     a multiple of PAGE_SIZE.  */
  #ifndef STACK_SIZE
! #define STACK_SIZE  (2 * 1024 * 1024)
  #endif
  
  /* The initial size of the thread stack.  Must be a multiple of PAGE_SIZE.  */
--- 339,345 ----
     THREAD_SELF implementation is used, this must be a power of two and
     a multiple of PAGE_SIZE.  */
  #ifndef STACK_SIZE
! #define STACK_SIZE  (128 * 1024)
  #endif
  
  /* The initial size of the thread stack.  Must be a multiple of PAGE_SIZE.  */
diff -r -c --exclude=*.info* glibc-2.2.4/linuxthreads/sysdeps/unix/sysv/linux/bits/local_lim.h glibc-2.2.4-new/linuxthreads/sysdeps/unix/sysv/linux/bits/local_lim.h
*** glibc-2.2.4/linuxthreads/sysdeps/unix/sysv/linux/bits/local_lim.h	Fri Jun  9 22:17:35 2000
--- glibc-2.2.4-new/linuxthreads/sysdeps/unix/sysv/linux/bits/local_lim.h	Tue Apr 16 15:10:38 2002
***************
*** 64,70 ****
  /* The number of threads per process.  */
  #define _POSIX_THREAD_THREADS_MAX	64
  /* This is the value this implementation supports.  */
! #define PTHREAD_THREADS_MAX	1024
  
  /* Maximum amount by which a process can descrease its asynchronous I/O
     priority level.  */
--- 64,70 ----
  /* The number of threads per process.  */
  #define _POSIX_THREAD_THREADS_MAX	64
  /* This is the value this implementation supports.  */
! #define PTHREAD_THREADS_MAX	4096
  
  /* Maximum amount by which a process can descrease its asynchronous I/O
     priority level.  */
diff -r -c --exclude=*.info* glibc-2.2.4/nss/nsswitch.c glibc-2.2.4-new/nss/nsswitch.c
*** glibc-2.2.4/nss/nsswitch.c	Mon Jul 23 20:54:48 2001
--- glibc-2.2.4-new/nss/nsswitch.c	Fri May  3 04:17:44 2002
***************
*** 496,501 ****
--- 496,502 ----
      {
        service_user *new_service;
        const char *name;
+       int name_alloc_len;
  
        while (isspace (line[0]))
  	++line;
***************
*** 510,522 ****
        if (name == line)
  	return result;
  
  
        new_service = (service_user *) malloc (sizeof (service_user)
! 					     + (line - name + 1));
        if (new_service == NULL)
  	return result;
  
!       *((char *) __mempcpy (new_service->name, name, line - name)) = '\0';
  
        /* Set default actions.  */
        new_service->actions[2 + NSS_STATUS_TRYAGAIN] = NSS_ACTION_CONTINUE;
--- 511,533 ----
        if (name == line)
  	return result;
  
+       name_alloc_len = line - name + 1;
+ 
+ #ifdef DO_STATIC_NSS
+       if (!((name_alloc_len == 6 && strncmp(name,"files",5) == 0) ||
+ 	    (name_alloc_len == 4 && strncmp(name,"dns",3) == 0)))
+       {
+ 	name = (char*) "files";
+         name_alloc_len=6;
+       }
+ #endif      
  
        new_service = (service_user *) malloc (sizeof (service_user)
! 					     + name_alloc_len);
        if (new_service == NULL)
  	return result;
  
!       *((char *) __mempcpy (new_service->name, name, name_alloc_len)) = '\0';
  
        /* Set default actions.  */
        new_service->actions[2 + NSS_STATUS_TRYAGAIN] = NSS_ACTION_CONTINUE;
diff -r -c --exclude=*.info* glibc-2.2.4/time/Makefile glibc-2.2.4-new/time/Makefile
*** glibc-2.2.4/time/Makefile	Fri Aug 10 22:12:07 2001
--- glibc-2.2.4-new/time/Makefile	Tue Apr 16 15:11:09 2002
***************
*** 37,44 ****
  
  include ../Rules
  
! tz-cflags = -DTZDIR='"$(zonedir)"' \
! 	    -DTZDEFAULT='"$(localtime-file)"' \
  	    -DTZDEFRULES='"$(posixrules-file)"'
  
  CFLAGS-tzfile.c = $(tz-cflags)
--- 37,44 ----
  
  include ../Rules
  
! tz-cflags = -DTZDIR='"/usr/share/zoneinfo/"' \
! 	    -DTZDEFAULT='"/etc/localtime"' \
  	    -DTZDEFRULES='"$(posixrules-file)"'
  
  CFLAGS-tzfile.c = $(tz-cflags)
diff -r -c --exclude=*.info* glibc-2.2.4/timezone/Makefile glibc-2.2.4-new/timezone/Makefile
*** glibc-2.2.4/timezone/Makefile	Mon Jul 23 20:58:05 2001
--- glibc-2.2.4-new/timezone/Makefile	Tue Apr 16 15:11:09 2002
***************
*** 159,166 ****
  
  $(objpfx)zic: $(objpfx)scheck.o $(objpfx)ialloc.o
  
! tz-cflags = -DTZDIR='"$(zonedir)"' \
! 	    -DTZDEFAULT='"$(localtime-file)"' \
  	    -DTZDEFRULES='"$(posixrules-file)"' \
  	    -DTM_GMTOFF=tm_gmtoff -DTM_ZONE=tm_zone
  
--- 159,166 ----
  
  $(objpfx)zic: $(objpfx)scheck.o $(objpfx)ialloc.o
  
! tz-cflags = -DTZDIR='"/usr/share/zoneinfo/"' \
! 	    -DTZDEFAULT='"/etc/localtime"' \
  	    -DTZDEFRULES='"$(posixrules-file)"' \
  	    -DTM_GMTOFF=tm_gmtoff -DTM_ZONE=tm_zone
  
